plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseMySqlConnectorTest.java

```diff
@@ -337,6 +343,107 @@ public void testPredicatePushdown()
+                // no pushdown because it was converted to range predicate
+                .isNotFullyPushedDown(
+                        node(
+                                FilterNode.class,
+                                // verify that no constraint is applied by the connector
+                                tableScan(
+                                        tableHandle -> ((JdbcTableHandle) tableHandle).getConstraint().isAll(),
+                                        TupleDomain.all(),
+                                        ImmutableMap.of())));
+
+        // varchar IN without domain compaction
+        assertThat(query(format("SELECT regionkey, nationkey, name FROM %s WHERE name IN ('POLAND', 'ROMANIA', 'VIETNAM')", objectName)))
+                .matches("VALUES " +
+                        "(BIGINT '3', BIGINT '19', CAST('ROMANIA' AS varchar(255))), " +
+                        "(BIGINT '2', BIGINT '21', CAST('VIETNAM' AS varchar(255)))")
+                .isFullyPushedDown();
+
+        // varchar IN with small compaction threshold
+        assertThat(query(
+                Session.builder(getSession())
+                        .setCatalogSessionProperty("mysql", "domain_compaction_threshold", "1")
+                        .build(),
+                format("SELECT regionkey, nationkey, name FROM %s WHERE name IN ('POLAND', 'ROMANIA', 'VIETNAM')", objectName)))
+                .matches("VALUES " +
+                        "(BIGINT '3', BIGINT '19', CAST('ROMANIA' AS varchar(255))), " +
+                        "(BIGINT '2', BIGINT '21', CAST('VIETNAM' AS varchar(255)))")
+                // no pushdown because it was converted to range predicate
+                .isNotFullyPushedDown(
+                        node(
+                                FilterNode.class,
+                                // verify that no constraint is applied by the connector
+                                tableScan(
+                                        tableHandle -> ((JdbcTableHandle) tableHandle).getConstraint().isAll(),
+                                        TupleDomain.all(),
+                                        ImmutableMap.of())));
+        // varchar different case
+        assertThat(query(format("SELECT regionkey, nationkey, name FROM %s WHERE name = 'romania'", objectName)))
+                .returnsEmptyResult()
+                .isFullyPushedDown();
+
+        Session joinPushdownEnabled = joinPushdownEnabled(getSession());
+        // join on varchar columns
+        assertThat(query(joinPushdownEnabled, format("SELECT n.name, n2.regionkey FROM %1$s n JOIN %1$s n2 ON n.name = n2.name", objectName)))
+                .joinIsNotFullyPushedDown();
+    }
+
+    @DataProvider
+    public static Object[][] collation()
+    {
+        return new Object[][] {{"latin1", "latin1_general_cs"}, {"utf8", "utf8_bin"}};
```

Comment:
These are the only collation options or ?

Reply:
There are a lot of them, I just took 2 of them as an example - ending with _cs and _bin, which are case sensitive.